lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2452,11 +2456,14 @@ protected static function get_property_value( $styles, $path, $theme_json = null
 		 * where the values is an array with a "ref" key, pointing to a path.
 		 * For example: { "ref": "style.color.background" } => "#fff".
 		 */
+
 		if ( is_array( $value ) && isset( $value['ref'] ) ) {
 			$value_path = explode( '.', $value['ref'] );
-			$ref_value  = _wp_array_get( $theme_json, $value_path );
+			$ref_value  = _wp_array_get( $theme_json, $value_path, null );
+			// Background Image refs can refer to a string or an array containing a URL string.
+			$ref_value_url = $ref_value['url'] ?? null;
```

Comment:
This is specifically for background images. 

It's part of the reason why I think this could be a style engine util, used both internally, and externally by `WP_Theme_JSON`.

The style engine could subsume many style-related utils and potentially reduce the site of `WP_Theme_JSON`.